src/persistence.erl

```diff
@@ -71,46 +98,30 @@ code_change(_OldVsn, State, _Extra) ->
 
 %% redis api
 
-get(ServerName, Key) ->
-    gen_server:call(ServerName, {get, Key}).
-
-set(ServerName, Key, Value) ->
-    gen_server:call(ServerName, {set, Key, Value}).
-
-del(ServerName, Key) ->
-    gen_server:call(ServerName, {del, Key}).
-
-keys(ServerName, Pattern) ->
-    gen_server:call(ServerName, {keys, Pattern}).
-
-%% persistence higher level api
-
-get_user(ServerName, Id) ->
-    get(ServerName, Id).
-
-get_users(ServerName) ->
-    keys(ServerName, "*").
-
-add_user(ServerName, Name) ->
-    Id = << <<Y>> ||<<X:4>> <= crypto:hash(md5, term_to_binary(make_ref())), Y <- integer_to_list(X,16)>>,
-    Json = json_utils:encode(#{<<"id">> => Id, <<"Name">> => Name}),
-    Response = set(ServerName, Id, Json),
-    case Response of
-        {error, Reason} -> {error, Reason};
-        {ok, _Result} -> {ok, Id}
-    end.
-
-update_user(ServerName, Id, NewName) ->
-    Json = json_utils:encode(#{<<"id">> => Id, <<"Name">> => NewName}),
-    Response = set(ServerName, Id, Json),
-    case Response of
-        {error, Reason} -> {error, Reason};
-        {ok, _Result} -> ok
-    end.
-
-delete_user(ServerName, Id) ->
-    Response = del(ServerName, Id),
-    case Response of
-        {error, Reason} -> {error, Reason};
-        {ok, _Result} -> ok
-    end.
\ No newline at end of file
+get(Key) ->
+    gen_server:call(?REDIS_SERVER, {get, Key}).
+
+set(Key, Value) ->
+    gen_server:call(?REDIS_SERVER, {set, Key, Value}).
+
+del(Key) ->
+    gen_server:call(?REDIS_SERVER, {del, Key}).
+
+keys(Pattern) ->
+    gen_server:call(?REDIS_SERVER, {keys, Pattern}).
+
+set_add(Key, Value) ->
+    gen_server:call(?REDIS_SERVER, {set_add, Key, Value}).
+
+set_remove(Key, Value) ->
+    gen_server:call(?REDIS_SERVER, {set_remove, Key, Value}).
+
+set_is_member(Key, Value) ->
+    gen_server:call(?REDIS_SERVER, {set_is_member, Key, Value}).
+
+set_list_members(Key) ->
+    gen_server:call(?REDIS_SERVER, {set_list_members, Key}).
+
+%%%% @todo Added By Pawel
+if_exists(Key) ->
```

Comment:
Przedrostek `if` sugeruje że to jakiś warunek, a to zwykłe sprawdzenie czy klucz istnieje. Jak już to `exists`

Reply:
Słusznie. Ale wewnętrzną strukturę modułów graph.erl i persistence.erl zostawiam wam, zgodnie z umową